vector/src/main/java/im/vector/riotx/features/crypto/quads/SharedSecureStorageViewModel.kt

```diff
@@ -117,11 +118,15 @@ class SharedSecureStorageViewModel @AssistedInject constructor(
                 withContext(Dispatchers.IO) {
                     args.requestedSecrets.forEach {
                         val res = awaitCallback<String> { callback ->
-                            session.sharedSecretStorageService.getSecret(
-                                    name = it,
-                                    keyId = keyInfo.id,
-                                    secretKey = keySpec,
-                                    callback = callback)
+                            if (session.getAccountDataEvent(it) != null) {
+                                session.sharedSecretStorageService.getSecret(
+                                        name = it,
+                                        keyId = keyInfo.id,
+                                        secretKey = keySpec,
+                                        callback = callback)
+                            } else {
+                                Timber.w("## Cannot find secret $it in SSSS, skip")
+                            }
                         }
                         decryptedSecretMap[it] = res
```

Comment:
If the else branch is hit, `awaitCallback` will suspend forever! Since `callback` is not being called in that branch.
This is probably what you wanted to do. (Unless you wanted to store null in the map)
```suggestion
                        if (session.getAccountDataEvent(it) != null) {
                            val res = awaitCallback<String> { callback ->
                                session.sharedSecretStorageService.getSecret(
                                    name = it,
                                    keyId = keyInfo.id,
                                    secretKey = keySpec,
                                    callback = callback)
                            }
                            decryptedSecretMap[it] = res
                        } else {
                            Timber.w("## Cannot find secret $it in SSSS, skip")
                        }
```

Reply:
thx good catch!